cheroot/connections.py

```diff
@@ -267,9 +267,11 @@ def close(self):
             conn.close()
         self._readable_conns.clear()
 
-        for _, key in self._selector.get_map().items():
-            if key.data != self.server:  # server closes its own socket
-                key.data.socket.close()
+        selector_map = self._selector.get_map()
+        if selector_map is not None:
+            for _, key in selector_map.items():
+                if key.data != self.server:  # server closes its own socket
+                    key.data.socket.close()
```

Comment:
It's cleaner to do try/except rather than have a deeply nested loop
```suggestion
        try:
            connections = (
                conn.data.socket for conn in self._selector.get_map().items()
                if conn.data != self.server  # server closes its own socket
            )
        except AttributeError:
            connections = ()

        for connection in connections:
            connection.close()
```

Reply:
I get the whole "ask for forgiveness bit", but in this case the trouble with that is that if a future update to e.g. the `selector` library renames `get_map` to, say, `map`, you'll still get an `AttributeError` but for the wrong reason and you'll just swallow the error without ever closing the server correctly.

I could split this up as
```python
selector_map = self._selector.get_map()
try:
    selector_items = selector_map.items()
except AttributeError:
    selector_items = ()

connections = (
    conn.data.socket for conn in selector_items
    if conn.data != self.server  # server closes its own socket
)

for connection in connections:
    connection.close()
```

But that seems overly verbose, to me. Let me know what you think, yltimately, I'll go with what you suggest.

Reply:
try/except is necessary, also it's unlikely that this library changes the public interface suddenly. Alternatively you could do something like `a_dict or {}.items()` to mitigate the `None` case. My main concern here is deep nesting.